File: Problem-004/solution.py
def largest_palindrome_product(n):
    max_num = 10**n - 1
    min_num = 10**(n - 1)
    largest = 0

    for i in range(max_num, min_num - 1, -1):
        for j in range(i, min_num - 1, -1):
            product = i * j

            if product <= largest:
                break

            if str(product) == str(product)[::-1]:
                largest = product

    return largest

File: Problem-004/solution.js
function largestPalindromeProduct(n) {
    let largest = 0;
    const max = 10 ** n - 1;
    const min = 10 ** (n - 1);

    function isPalindrome(num) {
        let str = num.toString();
        return str === str.split("").reverse().join("");
    }

    for (let i = max; i >= min; i--) {
        for (let j = i; j >= min; j--) {
            let product = i * j;

            if (product <= largest) break;

            if (isPalindrome(product)) {
                largest = product;
            }
        }
    }

    return largest;
}
